data_transfer/jobs/shared.py

```diff
@@ -26,3 +45,68 @@ def batch_upload_data() -> None:
                 record.is_uploaded = True
                 update_record(record)
             dmpy.rm_local_data(zip_path)
+
+
+def prepare_data_folders() -> None:
+    """
+    Checks folders present in 'data/upload/ are finished
+    and moves them into the upload folder in the format:
+
+        DEVICEID-PATIENTID-STARTWEAR-ENDWEAR
+    """
+
+    @dataclass
+    class taskObject:
+        patient_id: str
+        device_id: str
+        folder: Path
+        records: List[Record]
+
+    ignore_folders = unfinished_folders()
+    all_folders = [
+        (p.name, d.name)
+        for p in config.storage_vol.iterdir()
+        if p.is_dir()
+        for d in p.iterdir()
+        if d.is_dir()
+    ]
+    ready_folders = [f for f in all_folders if f not in ignore_folders]
+
+    to_process = [
+        taskObject(
+            folder[0],
+            folder[1],
+            (config.storage_vol / folder[0] / folder[1]),
+            get_records_in_folder(folder[0], folder[1]),
+        )
+        for folder in ready_folders
+    ]
+
```

Comment:
As discussed elsewhere, this could be simplified by working with the database directly and removes a need to read folder structures that may change. 

Instead, I would work with the DeviceID and PatientID since they create a composite key that represents the files to be uploaded. In this way, we don't need to read any files or look in folders. To do that, I would query the DB such that it returns a list of all records where `is_downloaded, is_processed` are True, i.e, they need to be uploaded but have not been downloaded?

I would then transform that output to the following as it represents each Record in each folder

```python
{
  PatientID1-DeviceID1: [{Record1}, ... {Record2}],
  PatientID1-DeviceID2: [{Record1}, ... {Record2}]
}
```

From this, you can then filter this dict of lists as follows: if `is_downloaded=True` is in any `Record`. That way, now your final dict contains all PatientID/DeviceIDs that have a list of fully downloaded files and are thus ready to upload.